### PR TITLE
Add more details to readme.

### PR DESCRIPTION
**Pull request by smarlowucf**
### What does this PR do? Why are we making this change?

- Add more details to readme.
- Move contributing to a separate doc.
- Add more info to contributing doc.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+- pyenchant
+- travis-sphinx
+- ghp-import
+
+## Contribution Checklist
+
+- All patches must be signed. [Signing Commits](#signing-commits)
+- All contributed code must conform to flake8. [Code Style](#code-style)
+- All new code contributions must be accompanied by a test.
+    - Tests must pass and coverage remain above 100%. [Unit & Integration Tests](#unit-&-integration-tests)
```
Comment (rjschwei) [inline CONTRIBUTING.md]: Hmm, "above 100%" makes really no sense. I presume that a code path that is tested multiple times pushes the coverage to a number greater than 100% but that is simply incorrect. I think this should be re-worded, maybe "Test coverage needs to remain at 100%"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+$ git tag -a v{version}
+# git tag -a v0.0.1
+
+# Create a message with the changes since last release and push tags.
+$ git push --tags
+```
+
+## Unit & Integration Tests
+
+All tests should pass and test coverage should remain above 100%.
```
Comment (rjschwei) [inline CONTRIBUTING.md]: dito

File: CONTRIBUTING.md (hunk 1 above)
Comment (smarlowucf) [reply CONTRIBUTING.md]: Copy and paste mistake.

**Comment by smarlowucf** [comment]
@rjschwei Good?

**Pull request merged**